user_interface/EbikeUIController/EbikeUIController.ino

```diff
@@ -98,11 +108,17 @@ void timerISR() {
   }
   previousRegenState = currentRegenState; // Update previous regen state
 
-  // Poll E-BRAKE_ACTIVATED pin (Digital read)
-  bool currentBrakeState = digitalRead(E_BRAKE_ENGAGED);
-  if (currentBrakeState == LOW) {
-    // As long as the brake is engaged, activate regen
-    handleThrottleInput(3, throttle, activatedRegen); // Ebrake request
+  // Poll E-BRAKE_ACTIVATED pin (Analog read)
+  int currentBrakeState = analogRead(E_BRAKE_ENGAGED);
+  if (currentBrakeState < -1) {
+    throttle = currentBrakeState;
+    if (regenMethod == 1) {
+      handleThrottleInput(REGEN, throttle, activatedRegen, DIGITAL); // Digital Ebrake request
+    } 
+    else if (regenMethod == 2) {
+      handleThrottleInput(REGEN, throttle, activatedRegen, ANALOG); // Analog Ebrake request
+    }
```

Comment:
This if statement is redundant. Instead of changing whether we pass in Digital or Analog depending on the value of regenMethod, we should have a single call that directly passes in regenMethod